sycl/include/CL/sycl/types.hpp

```diff
@@ -1818,9 +1818,7 @@ class SwizzleOp {
     return Tmp.template convert<convertT, roundingMode>();
   }
 
-  template <typename asT>
-  typename detail::enable_if_t<asT::getNumElements() == getNumElements(), asT>
```

Comment:
I think the right fix should allow only types with the same size.

Reply:
According to the specification the size of resulting vec must be the same as size of original vec. Hence, the main point of using as() function is to convert data type (with increase or decrease of number of elements). For example: vec/swizzle<uint8_t, 16> => vec<int32_t, 4>.  The swizzle vector class must have almost the same interface as vec class.
The only difference between their interfaces is in unimportant to this patch things, hence, as() function should work as described for vec class.

Reply:
Right, so I think `swizzle::as` method should check sizes similar to vec type.
I.e. have something similar to:
```
    static_assert((sizeof(*this) == sizeof(asT)),
                  "The new SYCL vec type must have the same storage size in "
                  "bytes as this SYCL vec");
    static_assert(
        detail::is_contained<asT, detail::gtl::vector_basic_list>::value,
        "asT must be SYCL vec of a different element type and "
        "number of elements specified by asT");
```

Reply:
So the check is in the as() function in vector class, why it should be added to as() function in swizzle vector class?

Reply:
Because swizzle method has the same pre-requisites. You shouldn't rely that they will be validated somewhere else.

Reply:
Added in new commit